cpu/msp430_common/ldscripts/msp430_common.ld

```diff
@@ -12,6 +12,10 @@ SECTIONS
     _srom = ORIGIN(ROM);
     _erom = ORIGIN(ROM) + LENGTH(ROM);
 
+    .flash_writable (NOLOAD) : ALIGN(4) {
+        *(.flash_writable)
+    } > ROM
```

Comment:
Wouldn't it make sense to alphabetically sort symbols? It would be nice to upgrade the firmware and keep configuration data from the previous firmware. If the symbols are sorted alphabetically, this could be feasible. (One still cannot drop, add, rename or resize configuration data areas. But one at least does not have to worry about sections being reordered.)

Reply:
Could you elaborate ? Which symbols exactly ? E.g. how would `cortexm_base.ld` look like sorted ?

Reply:
E.g. for XFA symbols are sorted alphabetically, so that e.g. `bar` will always come before `foo`.

See https://github.com/RIOT-OS/RIOT/blob/a76e4bdc114f055e8eadb18c97e1705f61b391d6/cpu/cortexm_common/ldscripts/cortexm_base.ld#L58

Similar, if we would like to have `config_bar` to always come before `config_foo` inside the flash are used for configuration stuff, we should also sort it. My fear is that otherwise a different linker might place them in other order.

Reply:
Ahhh now I understand. Yeah that is a good point. The [documentation](https://sourceware.org/binutils/docs/ld/Input-Section-Wildcards.html) for `SORT` says that it only sorts files and sections. Therefore for `bar` to be before `foo` one would need to declare `bar` with attribute .e.g `__attribute__((section(".flash_writable.1")` and `foo` with attribute  `__attribute__((section(".flash_writable.2")`. 

Therefore I guess defining a helper macro such as:
```c
#define _FLASH_WRITABLE(name) __attribute__((aligned(FLASHPAGE_SIZE))) __attribute__((section(".flash_writable". # name)))
```

Would make this work?

Reply:
Or possibly even this:
```c
#define FLASH_WRITABLE(name, size) \
    __attribute__((aligned(FLASHPAGE_SIZE))) \
    __attribute__((section(".flash_writable." #name))) \
    static const uint8_t name [size * FLASHPAGE_SIZE];
```

Reply:
Added the sorting of the sections inside `.flash_writable` and added the `FLASH_WRITABLE` macro to take advantage of the sorting.